Training24/Test.c

```diff
@@ -1,14 +1,193 @@
+   system ("cls");           // Windows-specific command to clear the screen
+#endif
+}
+
+void FilterChar (char* buffer, const char* temp) {
+   int j = 0;
+   for (int i = 0; temp[i] != '\0'; i++) if (isalnum (temp[i])) buffer[j++] = tolower (temp[i]); // Convert to lowercase and add to buffer
+   buffer[j] = '\0';
+}
+
+/// <summary>Function to print test results for string.</summary>
+void FormatResult (int numTests, const char** inputs, const char** actualResults, const int* pass) {
+   printf (YELLOW "------------------- Test Results for Palindrome Check --------------------\n" RESET"| Test Case | Input                           | Output            |Result|\n"
+      "|-----------|---------------------------------|-------------------|------|\n");
+   for (int i = 0; i < numTests; i++)
+      printf ("| %-9d | %-31s | %-17s | %6s |\n", i + 1, inputs[i], actualResults[i] ? actualResults[i] : "NULL", pass[i] ? GREEN "PASS" RESET : RED "FAIL" RESET);
+   printf ("\n");
+}
+
+/// <summary>Function to print test results for reverse number.</summary>
+void FormatResult2 (int numTests, int* inputs, int* actualResults, int* pass) {
+   printf (YELLOW "--------------------- Test Results for Reverse Number --------------------\n" RESET"| Test Case | Input                           | Output            |Result|\n"
+      "|-----------|---------------------------------|-------------------|------|\n");
+   for (int i = 0; i < numTests; i++) printf ("| %-9d | %-31d | %-17d | %8s |\n", i + 1, inputs[i], actualResults[i], pass[i] ? GREEN "PASS" RESET : RED "FAIL" RESET);
+   printf ("\n");
+}
+
+/// <summary>Test cases for Palindrome function.</summary>
+void TestPalindrome () {
+   struct {
+      const char* input;
+      const char* expected;
+   } tests[] = {
+       {"civic", "Palindrome"},
+       {"madam", "Palindrome"},
+       {"12321", "Palindrome"},
+       {"Was it a car or a cat I saw", "Palindrome"},
+       {"Hello, World!", "Not Palindrome"},
+       {"A man, a plan, a canal, Panama", "Palindrome"},
+       {"A", "Palindrome"},
+       {"ab", "Not Palindrome"},
+       {"Able was I ere I saw Elba", "Palindrome"},
+       {"a@a", "Palindrome"},
+       {"I did, did I?", "Palindrome"}
+   };
+   int numTests = sizeof (tests) / sizeof (tests[0]);
+   const char** input = malloc (numTests * sizeof (const char*));
+   char** actualResults = malloc (numTests * sizeof (char*));
+   int* pass = malloc (numTests * sizeof (int));
+   if (input == NULL || actualResults == NULL || pass == NULL) {
+      printf ("Memory allocation failed\n");
+      return;
+   }
+   for (int i = 0; i < numTests; ++i) {
+      input[i] = tests[i].input;
+      char* temp = malloc (strlen (tests[i].input) + 1);
+      if (temp == NULL) {
+         printf ("Memory allocation failed for temp\n");
+         return;
+      }
+      FilterChar (temp, tests[i].input);
+      bool result = IsPalindrome (temp);
+      actualResults[i] = result ? "Palindrome" : "Not Palindrome";

```

Comment:
Since result is used only in one place, you can call IsPalindrome() directly in the ternary expression and use the return value there.
```suggestion
      actualResults[i] = IsPalindrome (temp)? "Palindrome" : "Not Palindrome";

```